set.mm

```diff
@@ -25076,23 +25085,45 @@ yield an eliminable and weakly (that is, object-level) conservative
   $}
 
   ${
-    $d x z $.
+    $d x y $.  $d x z $.
     hbab.1 $e |- ( ph -> A. x ph ) $.
     $( Bound-variable hypothesis builder for a class abstraction.  (Contributed
-       by NM, 1-Mar-1995.) $)
+       by NM, 1-Mar-1995.)  Add disjoint variable condition to avoid ~ ax-13 .
+       (Revised by Gino Giotto, 20-Jan-2024.) $)
     hbab $p |- ( z e. { y | ph } -> A. x z e. { y | ph } ) $=
-      ( cv cab wcel wsb df-clab hbsb hbxfrbi ) DFACGHACDIBADCJACDBEKL $.
+      ( cv cab wcel wsb df-clab hbsbw hbxfrbi ) DFACGHACDIBADCJACDBEKL $.
   $}
 
   ${
     $d x z $.
+    hbabALT.1 $e |- ( ph -> A. x ph ) $.
+    $( Alternate version of ~ hbab , with less dv conditions, but requiring
+       ~ ax-13 .  (Contributed by NM, 1-Mar-1995.)
+       (Proof modification is discouraged.)  (New usage is discouraged.) $)
+    hbabALT $p |- ( z e. { y | ph } -> A. x z e. { y | ph } ) $=
+      ( cv cab wcel wsb df-clab hbsb hbxfrbi ) DFACGHACDIBADCJACDBEKL $.
+  $}
+
+  ${
+    $d x y $.  $d x z $.
     nfsab.1 $e |- F/ x ph $.
     $( Bound-variable hypothesis builder for a class abstraction.  (Contributed
-       by Mario Carneiro, 11-Aug-2016.) $)
+       by Mario Carneiro, 11-Aug-2016.)  Add disjoint variable condition to
+       avoid ~ ax-13 .  (Revised by Gino Giotto, 20-Jan-2024.) $)
     nfsab $p |- F/ x z e. { y | ph } $=
       ( cv cab wcel nf5ri hbab nf5i ) DFACGHBABCDABEIJK $.
   $}
 
+  ${
+    $d x z $.
+    nfsabALT.1 $e |- F/ x ph $.
+    $( Alternate version of ~ nfsab , with less dv conditions, but requiring
+       ~ ax-13 .  (Contributed by Mario Carneiro, 11-Aug-2016.)
+       (Proof modification is discouraged.)  (New usage is discouraged.) $)
```

Comment:
I would keep nfsab as the standard version (with fewer DV conditions), and add the version with an additional DV condition as nfsabv, as usual, with comment "Version of nfsab with an additional disjoint variable condition, which requires fewer axioms." (or: "which does not require ~ ax -13" if it saves only that axiom).

Reply:
(same with hbab and the new hbabv)

Reply:
> ... and same for the other ones

If your request is specific for `nfsab` and `hbab` then it's not a problem, because they are not used that much. If your suggestion was speaking in general terms (as I interpreted it from the last sentence) then not only I'm against it in principle, but I couldn't even commit it in this PR because of how large it would become.

Reply:
I haven't followed everything, but what I say is:

* do not change the labels of existing theorems,
* if you add a weakened form of an existing theorem obtained by adding one or several DV conditions, then use a suffix `v` or `vv` or `w` to its label.

This is what we have been doing for a long time and I do not see a reason to change.

Reply:
A relabeling cannot generate that much work and it is entirely automatic:
```
MM> SET WIDTH 9999
MM> READ set.mm
MM> SAVE PROOF * / NORMAL
MM> WRITE SOURCE set.mm
MM> ERASE
[in your text editor, do a basic search and replace " old_label " --> " new_label "; note
the spaces, and of course you may need a temporary label if you want to swap two labels]
MM> SET WIDTH 79
MM> READ set.mm
MM> SAVE PROOF * / COMPRESSED
MM> WRITE SOURCE set.mm / REWRAP
```

So, no, it would not complicate things much, and no time would be wasted, or maybe a handful of minutes top.

That said, if the existing stronger theorems are not used much any more and are not interesting in themselves, they may indeed be considered the variant while the ones you introduce here are considered the main ones.

> > do not change the labels of existing theorems
>
> The original theorems are not even needed at all and I would happy to just make them as OLD, the only reason they are kept with a different name is because I was asked to do so.

Then I'm very happy someone asked not to deprecate them.

Reply:
> A relabeling cannot generate that much work and it is entirely automatic:

No! The method you provided doesn't work. You have to use the minimizer. Why? Two reasons:

- First and most important: not every replacement is admitted. You have to check for dv violations, and those can create domino effects that mess up everything. The minimizer automatically checks for them, so that I don't have to worry about them. Can you imagine using your method and then do `verify proof *` and getting hundreds of dv violations as consequence? It's just impractical to work with.

- Second: I cannot provide a PR of an edited proof without dropping ax-13 to it. Or at least it would be frown upon. The thing is, most theorems depend on ax-13 by multiple routes, so if I just remove one, that theorem would still depend on ax-13 and a reviewer may ask "Why making this change at all?". So I have to be more clever about it, and in fact I'm currently doing minimizations of combinations of theorems.

> the existing stronger theorems are not used much any more

The existing theorems are not used at all, they are used only between each other. This PR nullify their usage the same way I did with the previous one https://github.com/metamath/set.mm/pull/3778.

> So, no, it would not complicate things much, and no time would be wasted, or maybe a handful of minutes top.

Of course it would. A PR with hundreds of shifted proofs is not small at all, and you know that.

Reply:
If the new labels have already been agreed, then fine, but just two remarks because it could help in the future, and save time:

No, a relabeling does not require the minimizer. (To arrive at this PR, you may have used a variety of tools, but once we arrived at this PR, the difference between our two proposals is a mere relabeling.)

No, it would not take much time to review: when you have to review a huge diff which was the result of an automatic script, you do not review manually the diff: you run the same script locally and you check that the diff is zero.

Reply:
> Which would generate a PR so big that...

See the second of my two remarks above.

> What script are you referring to?

See the first of my two remarks above (and my previous message in this thread).

By the way: this MR is good to merge: I made suggestions in this conversation, and not blocking requests. I don't know why you got suddenly so defensive. I already said my appreciation of your work on multiple topics, and in particular on this one of reducing dependences on ax-13, and I'm saying it again: this is going in the good direction and I hope you keep going.

Reply:
> By the way: this MR is good to merge: I made suggestions in this conversation, and not blocking requests. I don't know why you got suddenly so defensive. I already said my appreciation of your work on multiple topics, and in particular on this one of reducing dependences on ax-13, and I'm saying it again: this is going in the good direction and I hope you keep going.

In retrospect, maybe I overreacted. This project requires to keep track of a lot of information and I guess I perceived your comment as a threat to my workflow or as something that would have forced me to repeat some steps again (turns out, eradicating ax-13 in a collaborative environment is much harder than in a database where I have full control). But maybe my harsh response wasn't justified. I'm sorry.

For my own sanity, I would still prefer to keep renaming as a separate task, maybe I'll note it somewhere.